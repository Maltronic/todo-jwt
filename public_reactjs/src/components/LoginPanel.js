import PropTypes from 'prop-types';
import React, {Component} from 'react';

export default class LoginPanel extends Component {
    constructor(props) {
        super(props);
        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleChange = this.handleChange.bind(this);
        this.state = {
            username: '',
            password: ''
        };
    }

    handleSubmit = (event) => {
        event.preventDefault();
        return this.props.submitCallback(this.state.username, this.state.password);
    };

    handleChange = (event) => {
        this.setState({[event.target.name]: event.target.value});
    };

    render() {
        const content = (this.props.completedPhrase) ? this.displayCompleted() : this.displayForm();
        return (
            <form onSubmit={this.handleSubmit}
                  className="form-signin pull-right">
                <div><h4>{this.props.title}</h4></div>
                {content}
            </form>
        );
    }

    displayCompleted = () => (
        <div>{this.props.completedPhrase}</div>
    );

    displayForm = () => (
        <div>
            <div>
                <label>User name:</label><br/>
                <input type="text" name="username" placeholder="username..." value={this.state.username}
                       onChange={this.handleChange}/>
            </div>
            <div>
                <label>Password:</label><br/>
                <input type="password" name="password" placeholder="password..." value={this.state.password}
                       onChange={this.handleChange}/>
            </div>
            <input className="btn btn-primary" type="submit" value="submit"/>
        </div>
    );
}

LoginPanel.propTypes = {
    completedPhrase: PropTypes.string,
    submitCallback: PropTypes.func.isRequired,
    title: PropTypes.string
};
