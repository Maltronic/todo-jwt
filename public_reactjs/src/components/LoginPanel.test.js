import React from 'react';
import { shallow, mount } from 'enzyme';
import LoginPanel from "./LoginPanel";

describe('LoginPanel renders correctly', () => {
    let props = {
        completedPhrase: null,
        onSelected: null,
        submitCallback: null,
        title: "Login:"
    };

    beforeEach(() => {
        props.onSelected = jest.fn();
        props.submitCallback = jest.fn();
    });

    it('Matches Snapshot', () => {
        const wrapper = shallow(
            <LoginPanel {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    it("Can be submitted", () => {
        const wrapper = mount(<LoginPanel {...props} />);
        wrapper.find('[type="submit"]').get(0).click();
        expect(props.submitCallback).toHaveBeenCalledTimes(1);
    });
});

