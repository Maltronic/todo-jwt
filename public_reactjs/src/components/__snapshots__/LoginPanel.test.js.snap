// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPanel renders correctly Matches Snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <form
    className="form-signin pull-right"
    onSubmit={[Function]}
>
    <div>
        <h4>
            Login:
        </h4>
    </div>
    <div>
        <div>
            <label>
                User name:
            </label>
            <br />
            <input
                name="username"
                onChange={[Function]}
                placeholder="username..."
                type="text"
                value=""
            />
        </div>
        <div>
            <label>
                Password:
            </label>
            <br />
            <input
                name="password"
                onChange={[Function]}
                placeholder="password..."
                type="password"
                value=""
            />
        </div>
        <input
            className="btn btn-primary"
            type="submit"
            value="submit"
        />
    </div>
</form>,
  "nodes": Array [
    <form
      className="form-signin pull-right"
      onSubmit={[Function]}
>
      <div>
            <h4>
                  Login:
            </h4>
      </div>
      <div>
            <div>
                  <label>
                        User name:
                  </label>
                  <br />
                  <input
                        name="username"
                        onChange={[Function]}
                        placeholder="username..."
                        type="text"
                        value=""
                  />
            </div>
            <div>
                  <label>
                        Password:
                  </label>
                  <br />
                  <input
                        name="password"
                        onChange={[Function]}
                        placeholder="password..."
                        type="password"
                        value=""
                  />
            </div>
            <input
                  className="btn btn-primary"
                  type="submit"
                  value="submit"
            />
      </div>
</form>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <LoginPanel
        completedPhrase={null}
        onSelected={[Function]}
        submitCallback={[Function]}
        title="Login:"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": LoginPanel {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "displayCompleted": [Function],
        "displayForm": [Function],
        "handleChange": [Function],
        "handleSubmit": [Function],
        "props": Object {
          "completedPhrase": null,
          "onSelected": [Function],
          "submitCallback": [Function],
          "title": "Login:",
        },
        "refs": Object {},
        "state": Object {
          "password": "",
          "username": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <form
          className="form-signin pull-right"
          onSubmit={[Function]}
>
          <div>
                    <h4>
                              Login:
                    </h4>
          </div>
          <div>
                    <div>
                              <label>
                                        User name:
                              </label>
                              <br />
                              <input
                                        name="username"
                                        onChange={[Function]}
                                        placeholder="username..."
                                        type="text"
                                        value=""
                              />
                    </div>
                    <div>
                              <label>
                                        Password:
                              </label>
                              <br />
                              <input
                                        name="password"
                                        onChange={[Function]}
                                        placeholder="password..."
                                        type="password"
                                        value=""
                              />
                    </div>
                    <input
                              className="btn btn-primary"
                              type="submit"
                              value="submit"
                    />
          </div>
</form>,
        "_debugID": 2,
        "_renderedOutput": <form
          className="form-signin pull-right"
          onSubmit={[Function]}
>
          <div>
                    <h4>
                              Login:
                    </h4>
          </div>
          <div>
                    <div>
                              <label>
                                        User name:
                              </label>
                              <br />
                              <input
                                        name="username"
                                        onChange={[Function]}
                                        placeholder="username..."
                                        type="text"
                                        value=""
                              />
                    </div>
                    <div>
                              <label>
                                        Password:
                              </label>
                              <br />
                              <input
                                        name="password"
                                        onChange={[Function]}
                                        placeholder="password..."
                                        type="password"
                                        value=""
                              />
                    </div>
                    <input
                              className="btn btn-primary"
                              type="submit"
                              value="submit"
                    />
          </div>
</form>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <LoginPanel
    completedPhrase={null}
    onSelected={[Function]}
    submitCallback={[Function]}
    title="Login:"
/>,
}
`;
